Monika After Story/game/script-compliments.rpy

```diff
@@ -888,6 +889,174 @@ label mas_compliment_humor_3:
+            "I'm not sure you mean that, [player]...",
+            "I doubt you mean that, [player]...",
+            "I don't think you really mean that, [player]...",
+            "If only you really meant that, [player]...",
+            "...Why do I think you just don't mean it?",
+            "...Why do I think you're just saying that?",
+            "...I can't really believe in that, [player].",
+            "I don't think that's true, [player]."
+        ]
+    
+    $ absence_length = mas_getAbsenceLength()
+    $ mas_flagEVL("mas_compliment_missed21", "CMP", EV_FLAG_HFM)
+    
+    if mas_isMoniNormal(higher=True):
+        $ mas_gainAffection(2)
+        $ hugchance = 2
+
+        #Really short absences
+        if absence_length < datetime.timedelta(hours=1):
+            $ hugchance = 0
+
+        if absence_length >= datetime.timedelta(days=3):
+            $ missedchance = renpy.random.randint(1,3)
+            if missedchance == 1:
+                m 6wub "I missed you a ton, [player]!"
+            elif missedchance == 2:
+                m 6wub "I missed you a whole lot, [player]!"
+            else:
+                m 6wubfb "...I ree{w=0.1}ee{w=0.1}ee{w=0.1}ally missed you [player]!"
+            $ hugchance = 17
+            if absence_length >= datetime.timedelta(days=7):
+                $ hugchance = 50
+        else:
+            m 6eub "I missed you too, [player]!"
+
+        $ missed_quip = random.choice(missed_quips)
+        show monika 2hublb
+        $ renpy.say(m, missed_quip)
+        if mas_isMoniEnamored(higher=True) and renpy.random.randint(1,50) <= hugchance:
+            m 5rsa "Say, [player]."
+            m 5rsc "I was kinda hoping that..."
+            m 5fsc "You know, since it's been a little while..."
+            m 2fkblc "Could you give me a hug? I've been feeling pretty lonely while you were gone.{nw}"
+            $ _history_list.pop()
+            menu:
+                m "Could you give me a hug? I've been feeling pretty lonely while you were gone.{fast}"
+                "Okay.":
+                    call monika_holdme_prep(lullaby=MAS_HOLDME_NO_LULLABY, stop_music=True, disable_music_menu=True)
+                    call monika_holdme_start
+                    call monika_holdme_end
+
+                    m 2fkblb "Mmm... that was really nice, [player]."
+                    m 2ekbsa "You really know how to make a girl feel special!"
+                    m 7rublsdra "Anyway..."
+                    m 1esa "What shall we do today, [mas_get_player_nickname()]?"
+
+                    $ mas_gainAffection(0.75)#hug aff
+                    return
+                "Not right now.":
+                    #a lil sad
+                    $ mas_loseAffection()
+                    m 6ekc "...alright, [player]. Maybe later, then?"
+    #Base negative responses on monika_love label
+    elif mas_isMoniUpset():
+        $ mas_gainAffection()
+
+        if _mas_getAffection() <= -50:
```

Comment:
We do not use raw affection in our code, as you can see that function is private. Consider using high level API as you did elsewhere (`mas_isMoniUpset`/`mas_isMoniDis`/etc).

Reply:
I based this on line 3773 of the current release's "monika_love" label in script-topics.rpy, where this exact check is also made. Is that planned to change? 

I can change the check to remove the cynical quips if we want to avoid using this function, but I had intended for these to be two separate areas within the mas_isMoniUpset() state like the aforementioned label.

Reply:
Did not intend to mark prior comment resolved. Repeat

>I based this on line 3773 of the current release's "monika_love" label in script-topics.rpy, where this exact check is also made. Is that planned to change?

>I can change the check to remove the cynical quips if we want to avoid using this function, but I had intended for these to be two separate areas within the mas_isMoniUpset() state like the aforementioned label.

Reply:
That's rather an exception/old code. I'd say we should avoid doing that where we can.

Reply:
Understood. I'll remove the cynical quips, unless you know a better way of subdividing states?

Reply:
Unsure, I think there isn't one.

Reply:
Resolved in https://github.com/Monika-After-Story/MonikaModDev/pull/7419/commits/75bf48f7cf086946667604865627cfafadbe9e1c.